### ISSUE TITLE
[Unstable] [Modded] Afflictions defined later in the affliction.xml file cannot be conditional checked in a periodic effect within an affliction defined prior.

### ISSUE THREAD

**Issue by Emanthealmighty** [Regalis11/Barotrauma#5317]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
If we have affliction1 and affliction2, where affliction1 is defined before affliction2, and we try to check affliction2's strength within a status effect in a periodic effect inside affliction1, the conditional will always return false, as it is unable to check the strength of it.

**Version**
V0.1300.0.3 (Unstable)

**Comment by Regalis11** [comment]
I'm afraid I couldn't reproduce this. The status effects aren't instantiated until the affliction is applied on a character, so the order in which they're defined shouldn't make any difference.

I tried adding this to Afflictions.xml before radiation sickness, and it seems to be working correctly (reducing movement speed if radiation sickness is above 50%):

```
<Affliction
  name="Testaffliction"
  identifier="testaffliction"
  type="poison"
  limbspecific="false"
  indicatorlimb="Torso"
  showiconthreshold="0"
  karmachangeonapplied="-1"
  treatmentthreshold="25"
  maxstrength="100"
  grainburst="3">
  <!-- Cause burns depending on strength -->
  <Effect minstrength="0" maxstrength="100" mingrainstrength="0.0" maxgrainstrength="0.5">
    <StatusEffect target="Character" setvalue="true" SpeedMultiplier="0.3">
      <Conditional radiationsickness="gt 50"/>
    </StatusEffect>
  </Effect>
  <icon texture="Content/UI/MainIconsAtlas.png" sourcerect="896,768,128,128" color="195,136,60,255" origin="0,0"/>
</Affliction>
```

**Comment by Emanthealmighty** [comment]
After some further testing, it seems that status effects inside effects such as the affliction you provided DO work correctly as expected, however status effects inside of **periodic effects** that reference afflictions defined later in the file won't work. Moving the referenced affliction so that it is defined before will allow the periodic effect to properly work.

**Comment by Regalis11** [comment]
Thanks for the extra info, it didn't occur to me to test this with periodic effects! Fixed in https://github.com/Regalis11/Barotrauma-development/commit/8ded7c30d40a7664cf5ffa9dbf9e775ebbe5721f

**Comment by Emanthealmighty** [comment]
Fixed as of V0.1300.0.4.

**Comment by Rokvach** [comment]
Closing
